avocado/utils/software_manager.py

```diff
@@ -517,6 +537,95 @@ def provides(self, name):
+
+        try:
+            result = process.system_output(cmd)
+            for line in result.splitlines():
+                if line.startswith('Name'):
+                    full_name += "%s-" % line.split()[-1]
+                if line.startswith('Version'):
+                    full_name += "%s-" % line.split()[-1]
+                if line.startswith('Release'):
+                    full_name += "%s-" % line.split()[-1]
+                    return full_name
+        except process.CmdError, details:
+            log.error(details)
+            return ""
+
+    def build_dep(self, name):
+        """
+        Install build-dependencies for package [name]
+
+        :param name: name of the package
+
+        :return True: If build dependencies are installed properly
+        """
+        option = '-y --tolerant'
+        cmd = 'yum-builddep %s %s' % (option, name)
+
+        try:
+            process.system(cmd, sudo=True)
+            return True
+        except process.CmdError, details:
+            log.error(details)
+            return False
+
+    def source(self, name, dest_path):
+        """
+        Returns a ready-to-build source path for given package [name].
+
+        :param name: name of the package
+        :param dest_path: destination_path
+
+        :return final_dir: path of ready-to-build directory
+        """
+        path = tempfile.mkdtemp(prefix='avocado_')
+        src_path = os.environ['HOME']
+        build_option = "-bp"
+        final_dir = None
+        if dest_path is not None:
+            build_option += " --define '_builddir %s'" % dest_path
+        else:
+            log.error("Please provide a valid path")
+            return ""
+        pkg_list = ['yum-utils', 'rpm-build']
+        for pkg in pkg_list:
+            if not self.check_installed(pkg):
+                if not self.install(pkg):
+                    log.error("SoftwareManager (YumBackend) can't get packages"
+                              "with dependency resolution: Package 'yum-utils'"
+                              "could not be installed")
+                    return ""
+        try:
+            process.system('yumdownloader --source %s --destdir %s' % (name, path))
+            full_name = self.info(name)
+            src_rpm = '%s.src.rpm' % os.path.join(path, full_name)
+            if self.rpm_install(src_rpm):
+                if self.build_dep(name):
+                    process.system("rpmbuild %s %s/rpmbuild/SPECS/%s.spec"
+                                   % (build_option, src_path, name))
+                    final_dir = os.path.join(dest_path, os.listdir(dest_path)[0])
+                else:
+                    log.error("Installing build dependencies failed")
+                    return ""
+            else:
+                log.error("Installing source rpm failed")
+                return ""
+            return final_dir
+        except process.CmdError, details:
+            log.error(details)
+            return ""
```

Comment:
I'll need to do functional test, but by looking at it it looks quite good (except the use of `info`)